plugins/ptp_operator/metrics/manager.go

```diff
@@ -419,19 +420,21 @@ func (p *PTPEventManager) GenPTPEvent(ptpProfileName string, oStats *stats.Stats
 		return // do nothing for holdover
 	case ptp.FREERUN:
 		if lastClockState != ptp.HOLDOVER {
-			// within range
-			log.Infof(" publishing event for (profile %s) %s with last state %s and current clock state %s and offset %d for ( Max/Min Threshold %d/%d )",
-				ptpProfileName, eventResourceName, oStats.LastSyncState(), clockState, ptpOffset, threshold.MaxOffsetThreshold, threshold.MinOffsetThreshold)
-			p.PublishEvent(clockState, ptpOffset, eventResourceName, eventType) // change to locked
+			if lastClockState != ptp.FREERUN { // don't send event if last event was freerun
+				log.Infof(" publishing event for (profile %s) %s with last state %s and current clock state %s and offset %d for ( Max/Min Threshold %d/%d )",
+					ptpProfileName, eventResourceName, oStats.LastSyncState(), clockState, ptpOffset, threshold.MaxOffsetThreshold, threshold.MinOffsetThreshold)
+				p.PublishEvent(clockState, ptpOffset, eventResourceName, eventType) // change to locked
```

Comment:
remove the comment. it is change to freerun, not locaked

Reply:
addressed the issue and made few changes, please review again